mitiq/rem/inverse_confusion_matrix.py

```diff
@@ -14,38 +14,35 @@
 
 
 def sample_probability_vector(
-    probability_vector: npt.NDArray[np.float64], samples: int
-) -> List[Bitstring]:
+    probability_vector: Sequence[float], samples: int
+) -> list[str]:
```

Comment:
I am curious as to what's usually preferred when we specify the return type? 

I see you have used `list[str]` here but you could also have used `from typing import List` before specifying the return type as `List[str]`. 

Searching around for this shows that we [do not need to use the typing module](https://stackoverflow.com/a/62775680) for `>= Python 3.9`

Reply:
Didn't we have this exact same discussion with @vprusso in [another PR](https://github.com/unitaryfund/mitiq/pull/2347)? iirc there we learned that `list[str]` is the way to go.

Reply:
Yes, we did. Which is why I was asking if we need to put this on our todos. 

We still use things like `from typing import List` elsewhere.

Reply:
We don't need to be so proactive. The aliases are deprecated, but they are unlikely to be removed in the foreseeable future.

Reply:
Agreed, we should not take action until we need to on this. In the meantime lets try to use the builtins instead of `typing.List`, but it's not a big concern since they both achieve the same goal.

If we do migrate eventually, we will have to import a large majority of the types from `collections.abc` since many of the types we rely on (e.g. [`typing.Sequence`](https://docs.python.org/3/library/typing.html#typing.Sequence)) are deprecated in favor of coming from `collections.abc`.